graylog2-server/src/main/java/org/graylog2/periodical/LeaderPresenceCheckPeriodical.java

```diff
@@ -0,0 +1,162 @@
+                                    "has been fixed.", LeaderPresenceCheckPeriodical.class));
+                }
+            } else {
+                if (!timeWithoutLeader.isRunning()) {
+                    timeWithoutLeader.start();
+                }
+
+                getLogger().debug("No leader node found. Elapsed time without leader node: {}", timeWithoutLeader);
+
+                if (shouldWarnForNoLeader()) {
+                    Notification notification = notificationService.buildNow()
+                            .addNode(serverStatus.getNodeId().toString())
+                            .addType(Notification.Type.NO_LEADER)
+                            .addSeverity(Notification.Severity.URGENT);
+                    notificationService.publishIfFirst(notification);
+                }
+            }
+        } catch (Exception e) {
+            LOG.warn("Caught exception during check for presence of leader node.", e);
+        }
+    }
+
+    private boolean shouldWarnForNoLeader() {
+        if (configuration.getLeaderElectionMode() != LeaderElectionMode.AUTOMATIC) {
+            return true;
+        }
+        // In automatic leader election mode, there is an expected time window between a node giving up the leader lock
+        // and another node claiming the lock, where there is no leader in the cluster. This time window should not
+        // exceed the configured polling interval of the leader lock in a healthy cluster. So if a cluster is still
+        // without a leader after consecutive checks, we'll consider this an erroneous condition. This doesn't cover
+        // cases of rapid thrashing of the leader status, but it's hopefully good enough.
+        final Duration gracePeriod = configuration.getLeaderElectionLockPollingInterval().plusSeconds(1);
+        return timeWithoutLeader.elapsed().compareTo(gracePeriod) > 0;
+    }
+
+    private boolean fixNoLeaderNotification() {
+        // intentional non-short-circuit boolean operator to also remove legacy notification
+        //noinspection deprecation
+        return notificationService.fixed(notificationService.build().addType(Notification.Type.NO_MASTER)) |
+                notificationService.fixed(notificationService.build().addType(Notification.Type.NO_LEADER));
+    }
+
+    @Nonnull
+    @Override
+    protected Logger getLogger() {
+        return LOG;
+    }
+
+    @Override
+    public boolean runsForever() {
+        return false;
+    }
+
+    @Override
+    public boolean stopOnGracefulShutdown() {
+        return false;
+    }
+
+    @Override
+    public boolean leaderOnly() {
+        return false;
+    }
+
+    @Override
+    public boolean startOnThisNode() {
+        return true;
+    }
+
+    @Override
+    public boolean isDaemon() {
+        return true;
+    }
+
+    @Override
+    public int getInitialDelaySeconds() {
+        return 0;
+    }
+
+    @Override
+    public int getPeriodSeconds() {
+        return 1;
```

Comment:
Do you think we need to run this every second?

Reply:
No, this is just how it was before, so I kept the interval. Any preference for an alternative value?

Reply:
I talked with @thll offline and we decided to use a five-second interval instead of one second. That's still responsive.